airbrakes/mock/mock_imu.py

```diff
@@ -61,22 +65,40 @@ def __init__(
         # Starts the process that fetches data from the log file
         self._data_fetch_process = multiprocessing.Process(
             target=self._fetch_data_loop,
-            args=(
-                self._log_file_path,
-                real_time_simulation,
-                start_after_log_buffer,
-            ),
+            args=(self._log_file_path, real_time_simulation, start_after_log_buffer),
             name="Mock IMU Process",
         )
 
         # Makes a boolean value that is shared between processes
         self._running = multiprocessing.Value("b", False)
 
     @staticmethod
-    def _convert_invalid_fields(value):
-        if not value:
-            return None
-        return ast.literal_eval(value)  # Convert string to list
+    def _convert_invalid_fields(value) -> list:
+        """
+        Convert invalid fields to Python objects or None.
+        :param value: The value to convert.
+        :return: The converted value
+        """
+        return None if not value else ast.literal_eval(value)
+
+    @staticmethod
+    def _calculate_start_index(log_file_path: Path) -> int:
+        """
+        Calculate the start index based on log buffer size and time differences.
+        :param log_file_path: The path of the log file to read data from.
+        :return: The index where the log buffer ends.
+        """
+        chunk_size = LOG_BUFFER_SIZE + 1
+        for chunk in pd.read_csv(
```

Comment:
Let's add a new comment here saying that we process in chunks because it's faster to read a small chunk of the file where we expect the buffer index to be rather than to load the whole file into memory.